win32/select.c

```diff
@@ -34,18 +34,22 @@
  * - Calling this with NULL sets as a portable way to sleep with sub-second
  *   accuracy is not supported.
  * */
-PHPAPI int php_select(int max_fd, fd_set *rfds, fd_set *wfds, fd_set *efds, struct timeval *tv)
+PHPAPI int php_select(php_socket_t max_fd, fd_set *rfds, fd_set *wfds, fd_set *efds, struct timeval *tv)
 {
 	ULONGLONG ms_total, limit;
 	HANDLE handles[MAXIMUM_WAIT_OBJECTS];
-	int handle_slot_to_fd[MAXIMUM_WAIT_OBJECTS];
-	int n_handles = 0, i;
+	php_socket_t handle_slot_to_fd[MAXIMUM_WAIT_OBJECTS];
+	php_socket_t n_handles = 0, i;
```

Comment:
By having unsigned here, _get_osfhandle arg can be overflown, too. Looks like the descriptors dedicated loop might need some rework.